api/envoy/config/retry_policy/conditional_retry/v2/conditional_retry.proto

```diff
@@ -0,0 +1,55 @@
+syntax = "proto3";
+
+package envoy.config.retry_policy.conditional_retry.v2;
+
+import "envoy/api/v2/route/route_components.proto";
+
+import "udpa/annotations/migrate.proto";
+
+option java_package = "io.envoyproxy.envoy.config.retry_policy.conditional_retry.v2";
+option java_outer_classname = "ConditionalRetryProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_migrate).move_to_package =
+    "envoy.extensions.retry_policy.conditional_retry.v3";
+
+// [#protodoc-title: Conditional Retry Policy]
+// [#extension: envoy.retry_policy.conditional_retry]
+
+message ConditionalRetry {
+  message RetryCondition {
```

Comment:
Yay another matcher! :) 

Can we please use the normal pattern here where we have a oneof in here where each condition is independent, with embedded and/or as needed?

Reply:
Done